File: code_wars/8kyu/alternate/alternate.ts
export function toAlternatingCase(s: string): string {
  const isLower = (character: string): boolean => {
    return (
      character === character.toLowerCase() &&
      character !== character.toUpperCase()
    );
  };
  let result = "";
  let chars = [...s];

  for (let i in chars) {
    if (isLower(chars[i])) {
      result += chars[i].toUpperCase();
    } else {
      result += chars[i].toLowerCase();
    }
  }

  return result;
}

File: code_wars/8kyu/alternate/alternate.py
def to_alternating_case(string):
    result = ""
    for item in list(string):
        if item.isalpha():
            if item.islower():
                result += item.upper()
            elif item.isupper():
                result += item.lower()
        else:
            result += item

    return result


print("start\n")
print(to_alternating_case("aBc1"))
print("finish\n")
